### PR TITLE
[device doctor] Use homebrew bin

### PR DESCRIPTION
**Pull request by keyonghan**
M1 bots are installing some binaries to /opt/homebrew/bin, which is not visible to `swarming` by default. This PR checks that path to make sure binaries can run successfully, like `ideviceinstaller`.

Potentially solve: https://github.com/flutter/flutter/issues/98426#issuecomment-1063513050

### CODE REVIEW

File: device_doctor/lib/src/utils.dart [hunk 1]
```diff
@@ -129,3 +130,15 @@ void writeToFile(String results, File file) {
     ..writeAsStringSync(results);
   return;
 }
+
+/// Return Mac binary path.
+///
+/// For M1 bots, binaries like `ideviceinstaller` are installed under `kM1DefaultBinaryPath`,
+/// where they are not visible in `$PATH` by default.
+Future<String> getMacBinaryPath(String name, {ProcessManager processManager}) async {
+  String path = await eval('which', <String>[name], canFail: true, processManager: processManager);
```
[the remark this answers is not in the document]
Comment (keyonghan) [reply device_doctor/lib/src/utils.dart:139]: Good catch! Added an exception when no binary is found in either location.

File: device_doctor/lib/src/utils.dart [hunk 2]
```diff
@@ -129,3 +130,15 @@ void writeToFile(String results, File file) {
     ..writeAsStringSync(results);
   return;
 }
+
+/// Return Mac binary path.
+///
+/// For M1 bots, binaries like `ideviceinstaller` are installed under `kM1DefaultBinaryPath`,
```
[the remark this answers is not in the document]
Comment (keyonghan) [reply device_doctor/lib/src/utils.dart]: Done

File: device_doctor/lib/src/utils.dart [hunk 3]
```diff
@@ -28,6 +28,7 @@ const String kScreenSaverCheckKey = 'screensaver';
 const String kScreenRotationCheckKey = 'screen_rotation';
 const String kBatteryLevelCheckKey = 'battery_level';
 const String kBatteryTemperatureCheckKey = 'battery_temperature';
+const String kM1DefaultBinaryPath = '/opt/homebrew/bin';
```
[the remark this answers is not in the document]
Comment (keyonghan) [reply device_doctor/lib/src/utils.dart]: Done.

File: device_doctor/lib/src/ios_device.dart [hunk 4]
```diff
@@ -223,8 +223,9 @@ class IosDevice implements Device {
   Future<bool> uninstall_applications({ProcessManager processManager}) async {
     processManager ??= LocalProcessManager();
     String result;
+    final String ideviceinstaller = await getMacBinaryPath('ideviceinstaller', processManager: processManager);
```
[the remark this answers is not in the document]
Comment (keyonghan) [reply device_doctor/lib/src/ios_device.dart]: Done.

**Review by godofredoc — APPROVED** [review]

**Pull request merged**